src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1472,6 +1473,37 @@ def validate(self) -> None:
 yaml.add_representer(MDSSpec, ServiceSpec.yaml_representer)
 
 
+class MONSpec(ServiceSpec):
+    def __init__(self,
+                 service_type: str,
+                 service_id: Optional[str] = None,
+                 placement: Optional[PlacementSpec] = None,
+                 count: Optional[int] = None,
+                 config: Optional[Dict[str, str]] = None,
+                 unmanaged: bool = False,
+                 preview_only: bool = False,
+                 networks: Optional[List[str]] = None,
+                 extra_container_args: Optional[List[str]] = None,
+                 custom_configs: Optional[List[CustomConfig]] = None,
+                 crush_locations: Optional[Dict[str, List[str]]] = None,
+                 ):
+        assert service_type == 'mon'
+        super(MONSpec, self).__init__('mon', service_id=service_id,
+                                      placement=placement,
+                                      count=count,
+                                      config=config,
+                                      unmanaged=unmanaged,
+                                      preview_only=preview_only,
+                                      networks=networks,
+                                      extra_container_args=extra_container_args,
+                                      custom_configs=custom_configs)
+
+        self.crush_locations = crush_locations
```

Comment:
I added a validation function. It validates all the hostnames are valid hostnames (although not that the hosts are known hosts, which is on purpose) and that there is only one "=" in the crush loc. Since you can have custom bucket types and locations we can't do much more validation on them than that I think.